pkg/builtin/builtin.go

```diff
@@ -355,6 +360,36 @@ func (pw *progressWriter) Write(p []byte) (n int, err error) {
 	return len(p), nil
 }
 
+func getWorkspaceEnvFileContents(envs []string) ([]string, error) {
+	var envContents []string
+
+	dir, err := getWorkspaceDir(envs)
+	if err != nil {
+		return nil, err
+	}
+
+	file := filepath.Join(dir, "gptscript.env")
+
+	// Lock the file to prevent concurrent writes from other tool calls.
+	locker.RLock(file)
+	defer locker.RUnlock(file)
+
+	// This is optional, so no errors are returned if the file does not exist.
+	log.Debugf("Reading file %s", file)
+	data, err := os.ReadFile(file)
+	if errors.Is(err, fs.ErrNotExist) {
+		log.Debugf("The file %s does not exist", file)
+		return envContents, nil
+	} else if err != nil {
+		log.Debugf("Failed to read file %s: %v", file, err.Error())
+		return envContents, nil
+	}
+
+	envContents = append(envContents, strings.Split(string(data), "\n")...)
+
+	return envContents, nil
```

Comment:
nit: the `envContent` variable is not needed. You could do this:
```suggestion
	return strings.Split(string(data), "\n"), nil
```

Reply:
Also, do we want to filter out stuff that isn't in the form `FOO=bar`?